### PR TITLE
Add wrapper to author to avoid long name ui problem

### LINKED ISSUE #22866: Long full name/username causes repo view to overflow
### Description

Similar as #22360, but with full name/username.

I can't reproduce on the Gitea demo site because it returns "Bad Gateway".

### Screenshots

image

### Gitea Version

1.18.3

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

_No response_

### Browser Version

Chrome 109.0.5414.119

### PR DESCRIPTION
**Pull request by HesterG**
Close #22866. Main change is to add a `author-wrapper` class around author name, like the wrapper added to message. The `max-width` is set to 200px on PC, and 100px on mobile device for now. Which will work like below:

截屏2023-02-21 11 57 53

截屏2023-02-21 11 58 43

And `title` is added to the wrapper like it did in message wrapper. So the full author name will show on hover.

### CODE REVIEW

File: templates/repo/view_list.tmpl [hunk 1]
```diff
@@ -8,14 +8,14 @@
 					{{if .LatestCommitUser}}
 						{{avatar $.Context .LatestCommitUser 24}}
 						{{if .LatestCommitUser.FullName}}
-							<a class="muted" href="{{.LatestCommitUser.HomeLink}}"><strong>{{.LatestCommitUser.FullName}}</strong></a>
+							<span class="author-wrapper" title="{{.LatestCommitUser.FullName}}"><a class="muted" href="{{.LatestCommitUser.HomeLink}}"><strong>{{.LatestCommitUser.FullName}}</strong></a></span>
```
Comment (wxiaoguang) [inline templates/repo/view_list.tmpl]: Why not put these styles and title on `<a>` directly?

Comment (HesterG) [reply templates/repo/view_list.tmpl]: > Why not put these styles and title on `<a>` directly?

Yes that is better putting them directly on `<a>` tag here. Changed that.

File: templates/repo/view_list.tmpl [hunk 2]
```diff
@@ -8,14 +8,14 @@
 					{{if .LatestCommitUser}}
 						{{avatar $.Context .LatestCommitUser 24}}
 						{{if .LatestCommitUser.FullName}}
-							<a class="muted" href="{{.LatestCommitUser.HomeLink}}"><strong>{{.LatestCommitUser.FullName}}</strong></a>
+							<a class="muted author-wrapper" title="{{.LatestCommitUser.FullName}}" href="{{.LatestCommitUser.HomeLink}}"><strong>{{.LatestCommitUser.FullName}}</strong></a>
 						{{else}}
-							<a class="muted" href="{{.LatestCommitUser.HomeLink}}"><strong>{{if .LatestCommit.Author}}{{.LatestCommit.Author.Name}}{{else}}{{.LatestCommitUser.Name}}{{end}}</strong></a>
+							<a class="muted author-wrapper" title="{{if .LatestCommit.Author}}{{.LatestCommit.Author.Name}}{{else}}{{.LatestCommitUser.Name}}{{end}}" href="{{.LatestCommitUser.HomeLink}}"><strong>{{if .LatestCommit.Author}}{{.LatestCommit.Author.Name}}{{else}}{{.LatestCommitUser.Name}}{{end}}</strong></a></span>
```
Comment (wxiaoguang) [inline templates/repo/view_list.tmpl]: There is an extra `</span>`

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by Maks1mS** [comment]
It doesn't seem to fix the problem (сheck in other languages).
image

**Comment by wxiaoguang** [comment]
I think the answer would be "flex". These `max-width:` styles don't work well for such layout.

**Comment by HesterG** [comment]
> I think the answer would be "flex". These `max-width:` styles don't work well for such layout.

Flex could fix it, but might need more code changes like overriding of padding introduced by fomantic `<th>` and background change, etc. Used `table-layout:fixed` and reduced max-width for names for now.

**Comment by HesterG** [comment]
> It doesn't seem to fix the problem (сheck in other languages). image

Hi, tried another fix, please check again.

**Comment by Maks1mS** [comment]
> > It doesn't seem to fix the problem (сheck in other languages). image
> 
> Hi, tried another fix, please check again.

I think 525 px is the best option (including commit signing, CI status etc.)
image

**Comment by Maks1mS** [comment]
Also need to change in other media queries. I will write the optimal values ​​in the near future.

**Comment by Maks1mS** [comment]
```patch
diff --git a/web_src/less/_repository.less b/web_src/less/_repository.less
index e5f879e9e..5604386ba 100644
--- a/web_src/less/_repository.less
+++ b/web_src/less/_repository.less
@@ -2936,7 +2936,7 @@ tbody.commit-list {
   }
 
   th .message-wrapper {
-    max-width: 280px;
+    max-width: 120px;
   }
 }
 
@@ -2946,7 +2946,7 @@ tbody.commit-list {
   }
 
   th .message-wrapper {
-    max-width: 490px;
+    max-width: 350px;
   }
 }
 
@@ -2956,7 +2956,7 @@ tbody.commit-list {
   }
 
   th .message-wrapper {
-    max-width: 650px;
+    max-width: 525px;
   }
 }

```

Maybe not perfect, but I think it's a good option without using flex and a lot of code changes.

**Comment by HesterG** [comment]
> Maks1mS

added a commit to use these values

**Comment by HesterG** [comment]
> ```diff
> diff --git a/web_src/less/_repository.less b/web_src/less/_repository.less
> index e5f879e9e..5604386ba 100644
> --- a/web_src/less/_repository.less
> +++ b/web_src/less/_repository.less
> @@ -2936,7 +2936,7 @@ tbody.commit-list {
>    }
>  
>    th .message-wrapper {
> -    max-width: 280px;
> +    max-width: 120px;
>    }
>  }
>  
> @@ -2946,7 +2946,7 @@ tbody.commit-list {
>    }
>  
>    th .message-wrapper {
> -    max-width: 490px;
> +    max-width: 350px;
>    }
>  }
>  
> @@ -2956,7 +2956,7 @@ tbody.commit-list {
>    }
>  
>    th .message-wrapper {
> -    max-width: 650px;
> +    max-width: 525px;
>    }
>  }
> ```
> 
> Maybe not perfect, but I think it's a good option without using flex and a lot of code changes.

added a commit to use these values

**Comment by HesterG** [comment]
This is how the long name and long description look like on different size of screen:

截屏2023-02-23 16 01 47

截屏2023-02-23 16 01 59

截屏2023-02-23 16 02 06

截屏2023-02-23 16 02 24

Please feel free to check. And if no more change is required, i think this PR is ready to merge.

**Comment by jolheiser** [comment]
🎺 🤖

**Comment by yardenshoham** [comment]
Please send backport to v1.18

**Pull request merged**